lib/common/client/middleware/response/facility_parser.rb

```diff
@@ -18,129 +18,152 @@ def parse_body(env)
             path_test = env.url.path.match(/(FacilitySitePoint_\w{3})/) if path_test.nil?
             path_part = path_test[1]
             facility_map = facility_klass(path_part).attribute_map
-            env.body['features'].map { |location_data| build_facility_attributes(location_data, facility_map) }
+            env.body['features'].map do |location_data|
+              TempFacility.new(location_data, facility_map).build_facility_attributes
+            end
           end
 
-          def build_facility_attributes(location, mapping)
-            facility = { 'address' => {},
-                         'services' => {},
-                         'lat' => location['geometry']['y'],
-                         'long' => location['geometry']['x'] }
-            facility.merge!(make_direct_mappings(location, mapping))
-            facility.merge!(make_complex_mappings(location, mapping))
-            facility.merge!(make_address_mappings(location, mapping))
-            facility.merge!(map_benefits_services(location, mapping, facility['unique_id'])) if mapping['benefits']
-            facility.merge!(make_hours_mappings(location, mapping))
-            facility.merge!(map_health_services(location, mapping, facility['unique_id'])) if mapping['services']
-            facility
+          def facility_klass(path_part)
+            BaseFacility::PATHMAP[path_part]
           end
 
-          def make_direct_mappings(location, mapping)
-            %w[unique_id name classification website].each_with_object({}) do |name, attributes|
-              attributes[name] = strip(location['attributes'][mapping[name]])
-            end
-          end
+          class TempFacility
+            attr_accessor :facility, :location, :mapping
```

Comment:
The `TempFacility.facility` abstraction doesn't read very well. Can the classname or attribute name, be reworked to not collide with itself?

Reply:
Agreed! What do you think of `facility_hash` or maybe `temp_hash`? (My dream is to use the VHAFacility model object itself instead of this TempFacility class, but that refactor is too deep for this PR.)

Reply:
I mean, the facility subclasses

Reply:
So a "facility" is :
```
'address'
'services'
'lat'
'long'
```
?
That looks a lot like what I would call a "location" (an attribute that already exists). What about simply flattening out `TempFacility? e.g.:
```
TempFacility
  - location
  - mapping
  - address
  - services
   - geo_location
    - lat
    - long
```
I'm not that familiar with what data this service is communicating, so I value your judgement over mine. My only real objection is the `Facility.facility` conflict.

Reply:
The `facility` noted here has many more attributes than the ones you listed. Further confounding this, it is meant to represent multiple subtypes of facility, so there is no "one size fits all" object. The `facility` is meant to represent our format of facility, and the `location` is how the facility data is coming through from the GIS endpoint. The `facility_parser` translates the facility from gis and adds other things to the facility object, such as services. I think this warrants a heavier refactor at a later time. But I will rename the facility attribute for now.

Reply:
Would things be clearer if we renamed `location` -> `src_facility` and `facility` -> `dest_facility`?